GAE/src/org/akvo/flow/rest/ReportRestService.java

```diff
@@ -123,10 +123,17 @@ private boolean isAdminRole() {
     // find all reports belonging to the current user
     @RequestMapping(method = RequestMethod.GET, value = "")
     @ResponseBody
-    public Map<String, Object> listMyReports() {
+    public Map<String, Object> listMyReports(
+            @RequestParam(value = "reportType", defaultValue = "") String reportType
+            ) {
         final Map<String, Object> response = new HashMap<String, Object>();
         List<ReportDto> results = new ArrayList<ReportDto>();
-        List<Report> reports = reportDao.listAllByCurrentUser();
+        List<Report> reports;
+        if (reportType.equals("")) { //Cannot have null as the defaultValue
+            reportType = null;
```

Comment:
Why don't we just pass empty string and then check for empty in the method that uses it?

Reply:
Just for consistency. In most places we pass null to the dao methods to indicate a param should be ignored.